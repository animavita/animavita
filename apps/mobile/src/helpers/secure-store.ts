import * as SecureStore from 'expo-secure-store';

import { UserToken } from '../providers/auth-provider';

type KeyType = string;

const userTokenKey = 'userToken';

const getValueFor = async <T>(key: KeyType): Promise<T | null> => {
  const value = await SecureStore.getItemAsync(key);
  return value ? JSON.parse(value) : null;
};

const removeValueFor = async (key: KeyType) => {
  return await SecureStore.deleteItemAsync(key);
};

const save = async <T>(key: KeyType, value: T) => {
  const stringfiedValue = JSON.stringify(value);
  return await SecureStore.setItemAsync(key, stringfiedValue);
};

export const getUserToken = () => getValueFor<UserToken>(userTokenKey);
export const removeUserToken = () => removeValueFor(userTokenKey);
export const saveUserToken = (value: UserToken) => save(userTokenKey, value);
