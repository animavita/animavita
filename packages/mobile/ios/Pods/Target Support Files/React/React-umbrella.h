#ifdef __OBJC__
#import <UIKit/UIKit.h>
#else
#ifndef FOUNDATION_EXPORT
#if defined(__cplusplus)
#define FOUNDATION_EXPORT extern "C"
#else
#define FOUNDATION_EXPORT extern
#endif
#endif
#endif

#import "ARTCGFloatArray.h"
#import "ARTContainer.h"
#import "ARTGroup.h"
#import "ARTNode.h"
#import "ARTRenderable.h"
#import "ARTShape.h"
#import "ARTSurfaceView.h"
#import "ARTText.h"
#import "ARTTextFrame.h"
#import "ARTBrush.h"
#import "ARTLinearGradient.h"
#import "ARTPattern.h"
#import "ARTRadialGradient.h"
#import "ARTSolidColor.h"
#import "RCTConvert+ART.h"
#import "ARTGroupManager.h"
#import "ARTNodeManager.h"
#import "ARTRenderableManager.h"
#import "ARTShapeManager.h"
#import "ARTSurfaceViewManager.h"
#import "ARTTextManager.h"
#import "RCTAssert.h"
#import "RCTBridge+Private.h"
#import "RCTBridge.h"
#import "RCTBridgeDelegate.h"
#import "RCTBridgeMethod.h"
#import "RCTBridgeModule.h"
#import "RCTBundleURLProvider.h"
#import "RCTConvert.h"
#import "RCTCxxConvert.h"
#import "RCTDefines.h"
#import "RCTDisplayLink.h"
#import "RCTErrorCustomizer.h"
#import "RCTErrorInfo.h"
#import "RCTEventDispatcher.h"
#import "RCTFrameUpdate.h"
#import "RCTImageSource.h"
#import "RCTInvalidating.h"
#import "RCTJavaScriptExecutor.h"
#import "RCTJavaScriptLoader.h"
#import "RCTJSStackFrame.h"
#import "RCTKeyCommands.h"
#import "RCTLog.h"
#import "RCTManagedPointer.h"
#import "RCTModuleData.h"
#import "RCTModuleMethod.h"
#import "RCTMultipartDataTask.h"
#import "RCTMultipartStreamReader.h"
#import "RCTNullability.h"
#import "RCTParserUtils.h"
#import "RCTPerformanceLogger.h"
#import "RCTPlatform.h"
#import "RCTReloadCommand.h"
#import "RCTRootContentView.h"
#import "RCTRootView.h"
#import "RCTRootViewDelegate.h"
#import "RCTRootViewInternal.h"
#import "RCTTouchEvent.h"
#import "RCTTouchHandler.h"
#import "RCTURLRequestDelegate.h"
#import "RCTURLRequestHandler.h"
#import "RCTUtils.h"
#import "RCTVersion.h"
#import "RCTSurface.h"
#import "RCTSurfaceDelegate.h"
#import "RCTSurfaceRootShadowView.h"
#import "RCTSurfaceRootShadowViewDelegate.h"
#import "RCTSurfaceRootView.h"
#import "RCTSurfaceStage.h"
#import "RCTSurfaceView+Internal.h"
#import "RCTSurfaceView.h"
#import "RCTSurfaceHostingProxyRootView.h"
#import "RCTSurfaceHostingView.h"
#import "RCTSurfaceSizeMeasureMode.h"
#import "RCTAccessibilityManager.h"
#import "RCTAlertManager.h"
#import "RCTAppState.h"
#import "RCTAsyncLocalStorage.h"
#import "RCTClipboard.h"
#import "RCTDeviceInfo.h"
#import "RCTDevSettings.h"
#import "RCTEventEmitter.h"
#import "RCTExceptionsManager.h"
#import "RCTI18nManager.h"
#import "RCTI18nUtil.h"
#import "RCTKeyboardObserver.h"
#import "RCTLayoutAnimation.h"
#import "RCTLayoutAnimationGroup.h"
#import "RCTRedBox.h"
#import "RCTRedBoxExtraDataViewController.h"
#import "RCTSourceCode.h"
#import "RCTStatusBarManager.h"
#import "RCTTiming.h"
#import "RCTUIManager.h"
#import "RCTUIManagerObserverCoordinator.h"
#import "RCTUIManagerUtils.h"
#import "RCTFPSGraph.h"
#import "RCTMacros.h"
#import "RCTProfile.h"
#import "RCTUIUtils.h"
#import "RCTActivityIndicatorView.h"
#import "RCTActivityIndicatorViewManager.h"
#import "RCTAnimationType.h"
#import "RCTAutoInsetsProtocol.h"
#import "RCTBorderDrawing.h"
#import "RCTBorderStyle.h"
#import "RCTComponent.h"
#import "RCTComponentData.h"
#import "RCTConvert+CoreLocation.h"
#import "RCTConvert+Transform.h"
#import "RCTDatePicker.h"
#import "RCTDatePickerManager.h"
#import "RCTFont.h"
#import "RCTLayout.h"
#import "RCTMaskedView.h"
#import "RCTMaskedViewManager.h"
#import "RCTModalHostView.h"
#import "RCTModalHostViewController.h"
#import "RCTModalHostViewManager.h"
#import "RCTModalManager.h"
#import "RCTPicker.h"
#import "RCTPickerManager.h"
#import "RCTPointerEvents.h"
#import "RCTProgressViewManager.h"
#import "RCTRefreshControl.h"
#import "RCTRefreshControlManager.h"
#import "RCTRootShadowView.h"
#import "RCTSegmentedControl.h"
#import "RCTSegmentedControlManager.h"
#import "RCTShadowView+Internal.h"
#import "RCTShadowView+Layout.h"
#import "RCTShadowView.h"
#import "RCTSlider.h"
#import "RCTSliderManager.h"
#import "RCTSwitch.h"
#import "RCTSwitchManager.h"
#import "RCTTextDecorationLineType.h"
#import "RCTView.h"
#import "RCTViewManager.h"
#import "RCTWebView.h"
#import "RCTWebViewManager.h"
#import "RCTWKWebView.h"
#import "RCTWKWebViewManager.h"
#import "RCTWrapperViewController.h"
#import "RCTSafeAreaShadowView.h"
#import "RCTSafeAreaView.h"
#import "RCTSafeAreaViewLocalData.h"
#import "RCTSafeAreaViewManager.h"
#import "RCTScrollableProtocol.h"
#import "RCTScrollContentShadowView.h"
#import "RCTScrollContentView.h"
#import "RCTScrollContentViewManager.h"
#import "RCTScrollView.h"
#import "RCTScrollViewManager.h"
#import "UIView+Private.h"
#import "UIView+React.h"
#import "RCTDevLoadingView.h"
#import "RCTDevMenu.h"
#import "RCTInspectorDevServerHelper.h"
#import "RCTPackagerClient.h"
#import "RCTPackagerConnection.h"
#import "RCTInspector.h"
#import "RCTInspectorPackagerConnection.h"
#import "RCTActionSheetManager.h"
#import "RCTAnimationDriver.h"
#import "RCTDecayAnimation.h"
#import "RCTEventAnimation.h"
#import "RCTFrameAnimation.h"
#import "RCTSpringAnimation.h"
#import "RCTAdditionAnimatedNode.h"
#import "RCTAnimatedNode.h"
#import "RCTDiffClampAnimatedNode.h"
#import "RCTDivisionAnimatedNode.h"
#import "RCTInterpolationAnimatedNode.h"
#import "RCTModuloAnimatedNode.h"
#import "RCTMultiplicationAnimatedNode.h"
#import "RCTPropsAnimatedNode.h"
#import "RCTStyleAnimatedNode.h"
#import "RCTSubtractionAnimatedNode.h"
#import "RCTTrackingAnimatedNode.h"
#import "RCTTransformAnimatedNode.h"
#import "RCTValueAnimatedNode.h"
#import "RCTAnimationUtils.h"
#import "RCTNativeAnimatedModule.h"
#import "RCTNativeAnimatedNodesManager.h"
#import "RCTBlobManager.h"
#import "RCTFileReaderModule.h"
#import "RCTAssetsLibraryRequestHandler.h"
#import "RCTCameraRollManager.h"
#import "RCTImagePickerManager.h"
#import "RCTPhotoLibraryImageLoader.h"
#import "RCTLocationObserver.h"
#import "RCTGIFImageDecoder.h"
#import "RCTImageBlurUtils.h"
#import "RCTImageCache.h"
#import "RCTImageEditingManager.h"
#import "RCTImageLoader.h"
#import "RCTImageShadowView.h"
#import "RCTImageStoreManager.h"
#import "RCTImageUtils.h"
#import "RCTImageView.h"
#import "RCTImageViewManager.h"
#import "RCTLocalAssetImageLoader.h"
#import "RCTResizeMode.h"
#import "RCTLinkingManager.h"
#import "RCTDataRequestHandler.h"
#import "RCTFileRequestHandler.h"
#import "RCTHTTPRequestHandler.h"
#import "RCTNetInfo.h"
#import "RCTNetworking.h"
#import "RCTNetworkTask.h"
#import "RCTSettingsManager.h"
#import "RCTBaseTextShadowView.h"
#import "RCTBaseTextViewManager.h"
#import "RCTRawTextShadowView.h"
#import "RCTRawTextViewManager.h"
#import "RCTConvert+Text.h"
#import "RCTTextAttributes.h"
#import "RCTTextTransform.h"
#import "NSTextStorage+FontScaling.h"
#import "RCTTextShadowView.h"
#import "RCTTextView.h"
#import "RCTTextViewManager.h"
#import "RCTMultilineTextInputView.h"
#import "RCTMultilineTextInputViewManager.h"
#import "RCTUITextView.h"
#import "RCTBackedTextInputDelegate.h"
#import "RCTBackedTextInputDelegateAdapter.h"
#import "RCTBackedTextInputViewProtocol.h"
#import "RCTBaseTextInputShadowView.h"
#import "RCTBaseTextInputView.h"
#import "RCTBaseTextInputViewManager.h"
#import "RCTInputAccessoryShadowView.h"
#import "RCTInputAccessoryView.h"
#import "RCTInputAccessoryViewContent.h"
#import "RCTInputAccessoryViewManager.h"
#import "RCTTextSelection.h"
#import "RCTSinglelineTextInputView.h"
#import "RCTSinglelineTextInputViewManager.h"
#import "RCTUITextField.h"
#import "RCTVirtualTextShadowView.h"
#import "RCTVirtualTextViewManager.h"
#import "RCTVibration.h"
#import "RCTReconnectingWebSocket.h"
#import "RCTSRWebSocket.h"
#import "RCTWebSocketExecutor.h"
#import "RCTWebSocketModule.h"
#import "fishhook.h"

FOUNDATION_EXPORT double ReactVersionNumber;
FOUNDATION_EXPORT const unsigned char ReactVersionString[];

