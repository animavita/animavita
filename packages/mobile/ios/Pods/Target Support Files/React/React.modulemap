framework module React {
  umbrella header "React-umbrella.h"

  export *
  module * { export * }
}
