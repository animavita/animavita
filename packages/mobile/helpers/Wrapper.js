import React from 'react';
import { Provider } from 'react-redux';
import { MockedProvider } from '@apollo/react-testing';

export default ({ children, apolloStore, reduxStore }) => (
  <MockedProvider mocks={apolloStore}>
    <Provider store={reduxStore}>{children}</Provider>
  </MockedProvider>
);
