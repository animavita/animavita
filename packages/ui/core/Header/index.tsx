import React from 'react';
import styled from 'styled-components/native';
import FontAwesomeIcons from 'react-native-vector-icons/FontAwesome';
import {px2ddp, heightPercentageToDP, useTheme} from '@animavita/theme';

import Typography from '../Typography';

const Wrapper = styled.View`
  margin: ${heightPercentageToDP('5%')}px 0 0 ${px2ddp(10)}px;
  flex-direction: row;
  width: 100%;
`;

const BackButton = styled.TouchableOpacity.attrs({
  activeOpacity: 0.6,
  hitSlop: {top: 10, bottom: 10, left: 10, right: 10},
})`
  margin-right: ${px2ddp(10)}px;
  align-items: flex-start;
  justify-content: center;
`;

interface HeaderProps {
  title: string;
  onPrevious?: () => void;
}

const Header: React.FC<HeaderProps> = ({title, onPrevious}) => {
  const {theme} = useTheme();

  return (
    <Wrapper>
      {onPrevious && (
        <BackButton testID="backButton" onPress={onPrevious}>
          <FontAwesomeIcons name="chevron-left" size={18} color={theme.colors.text} />
        </BackButton>
      )}
      <Typography variant="title-3" type="bold">
        {title}
      </Typography>
    </Wrapper>
  );
};

export default Header;
