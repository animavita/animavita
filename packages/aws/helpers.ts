import * as SSM from '@aws-cdk/aws-ssm';
import * as CDK from '@aws-cdk/core';
import * as SecretsManager from '@aws-cdk/aws-secretsmanager';

export const getParam = (scope: CDK.Construct, name: string) => {
  return SSM.StringParameter.valueForStringParameter(scope, name);
};

export class ModeStack extends CDK.Stack {
  public readonly mode: string = this.node.tryGetContext('mode') || 'development';
  public readonly Mode: string =
    this.node.tryGetContext('mode').replace(/^\w/, (c: string) => c.toUpperCase()) || 'Development';

  constructor(scope: CDK.Construct, id: string) {
    super(scope, id);
  }
}

export function getEnvironmentVariables(stack: CDK.Stack, secretArn: string, variables: string[]) {
  const environmentSecret = SecretsManager.Secret.fromSecretArn(stack, 'AnimavitaSecrets', secretArn);

  const environment: {[key: string]: string} = {};
  for (const key of variables) {
    environment[key] = environmentSecret.secretValueFromJson(key).toString();
  }

  return environment;
}
