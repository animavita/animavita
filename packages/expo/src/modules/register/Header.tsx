import React from 'react';
import {useNavigation} from '@react-navigation/native';
import {TouchableOpacity} from 'react-native-gesture-handler';
import {AntDesign} from '@expo/vector-icons';
import {Typography} from '@animavita/ui/core';
import {useI18n} from '@animavita/i18n';

import {HeaderView, TitleImageView, BackIcon} from './styles';

const TITLE_2 = 'title-2';

const Header: React.FC = () => {
  const {t} = useI18n(['register']);
  const navigation = useNavigation();

  return (
    <HeaderView>
      <TouchableOpacity onPress={navigation.goBack}>
        <BackIcon name="arrowleft" size={24} />
      </TouchableOpacity>
      <TitleImageView>
        <Typography variant={TITLE_2} type={'bold'}>
          {t('header')}
        </Typography>
      </TitleImageView>
    </HeaderView>
  );
};

export default Header;
