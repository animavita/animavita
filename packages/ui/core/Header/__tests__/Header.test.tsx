import React from 'react';
import {fireEvent} from '@testing-library/react-native';
import renderer from 'react-test-renderer';

import {Mount, ThemeContext} from '../../../tests/helpers';
import Header from '../index';

const defaultProps = {
  title: 'Header',
  onPrevious: jest.fn().mockName('onPrevious'),
};

const mountFactory = propOverrides => {
  const headerProps = {...defaultProps, ...propOverrides};

  const element = Mount(<Header {...headerProps} />);

  return element;
};

describe('Header component', () => {
  beforeEach(() => {
    jest.clearAllMocks();
  });

  describe('renders correctly', () => {
    it('match snapshot', () => {
      const tree = renderer
        .create(ThemeContext(<Header title={defaultProps.title} onPrevious={defaultProps.onPrevious} />))
        .toJSON();
      expect(tree).toMatchSnapshot();
    });

    it('render title', () => {
      const {getByText} = mountFactory({title: 'Header'});

      expect(getByText('Header')).toBeTruthy();
    });
  });

  it('calls onPrevious', () => {
    const cb = jest.fn();
    const {getByTestId} = mountFactory({title: 'Header', onPrevious: cb});

    fireEvent.press(getByTestId('backButton'));

    expect(cb).toHaveBeenCalled();
  });
});
