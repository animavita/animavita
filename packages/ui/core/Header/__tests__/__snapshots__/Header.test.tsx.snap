// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Header component renders correctly 1`] = `
<View
  style={
    Array [
      Object {
        "flexDirection": "row",
        "marginBottom": 0,
        "marginLeft": 20,
        "marginRight": 0,
        "marginTop": 66.5,
        "width": "100%",
      },
    ]
  }
>
  <View
    accessible={true}
    focusable={true}
    hitSlop={
      Object {
        "bottom": 10,
        "left": 10,
        "right": 10,
        "top": 10,
      }
    }
    onClick={[Function]}
    onResponderGrant={[Function]}
    onResponderMove={[Function]}
    onResponderRelease={[Function]}
    onResponderTerminate={[Function]}
    onResponderTerminationRequest={[Function]}
    onStartShouldSetResponder={[Function]}
    style={
      Object {
        "alignItems": "flex-start",
        "justifyContent": "center",
        "marginRight": 20,
        "opacity": 1,
      }
    }
    testID="backButton"
  >
    <Text />
  </View>
  <Text
    style={
      Array [
        Object {
          "color": "#353434",
        },
        Object {
          "fontSize": 40,
          "fontWeight": "bold",
        },
      ]
    }
    themeName="light"
    type="bold"
  >
    Header
  </Text>
</View>
`;

exports[`Header component renders correctly match snapshot 1`] = `
<View
  style={
    Array [
      Object {
        "flexDirection": "row",
        "marginBottom": 0,
        "marginLeft": 20,
        "marginRight": 0,
        "marginTop": 66.5,
        "width": "100%",
      },
    ]
  }
>
  <View
    accessible={true}
    focusable={true}
    hitSlop={
      Object {
        "bottom": 10,
        "left": 10,
        "right": 10,
        "top": 10,
      }
    }
    onClick={[Function]}
    onResponderGrant={[Function]}
    onResponderMove={[Function]}
    onResponderRelease={[Function]}
    onResponderTerminate={[Function]}
    onResponderTerminationRequest={[Function]}
    onStartShouldSetResponder={[Function]}
    style={
      Object {
        "alignItems": "flex-start",
        "justifyContent": "center",
        "marginRight": 20,
        "opacity": 1,
      }
    }
    testID="backButton"
  >
    <Text />
  </View>
  <Text
    style={
      Array [
        Object {
          "color": "#353434",
        },
        Object {
          "fontSize": 40,
          "fontWeight": "bold",
        },
      ]
    }
    themeName="light"
    type="bold"
  >
    Header
  </Text>
</View>
`;
